app/jobs/modsulator_job.rb

```diff
@@ -121,6 +113,32 @@ def update_metadata(druid, xml_string, log)
     end
   end
 
+  # Open a new version for the given object if it is in the accessioned state.
+  def version_object(dor_object, original_filename, user_login, log)
+    if(accessioned(dor_object))
+      if !DorObjectWorkflowStatus.new(dor_object.pid).can_open_version?
+        log.puts("argo.bulk_metadata.bulk_log_unable_to_version #{dor_object.pid}")  # totally unexpected
+        return
+      end
+      commit_new_version(dor_object, original_filename)
+    end
+  end
+
+  def commit_new_version(dor_object, original_filename, user_login)
+    vers_md_upd_info = {
+      :significance => 'minor',
+      :description => "Descriptive metadata upload from #{original_filename}",
+      :opening_user_name => user_login
+    }
+    dor_object.open_new_version({:vers_md_upd_info => vers_md_upd_info})
+  end
+
+  def accessioned(dor_object)
+    return (dor_object.status_info[:status_code] == 6 ||
+            dor_object.status_info[:status_code] == 7 ||
+            dor_object.status_info[:status_code] == 8)
```

Comment:
the dor-services code that results in this issue is actually the dor-services code i mentioned on chat recently as an example of primitive obsession that could stand to be fixed (and the primitive obsession is at least a result of refactoring it to be more modular and reusable by different callers).